server/src/main/java/org/elasticsearch/common/compress/CompressedXContent.java

```diff
@@ -41,43 +43,43 @@
     private static final ThreadLocal<InflaterAndBuffer> inflater1 = ThreadLocal.withInitial(InflaterAndBuffer::new);
     private static final ThreadLocal<InflaterAndBuffer> inflater2 = ThreadLocal.withInitial(InflaterAndBuffer::new);
 
-    private static int crc32(BytesReference data) {
-        CRC32 crc32 = new CRC32();
+    private static String sha256(BytesReference data) {
+        MessageDigest messageDigest = MessageDigests.sha256();
         try {
-            data.writeTo(new CheckedOutputStream(Streams.NULL_OUTPUT_STREAM, crc32));
+            data.writeTo(new DigestOutputStream(Streams.NULL_OUTPUT_STREAM, messageDigest));
         } catch (IOException bogus) {
             // cannot happen
             throw new Error(bogus);
         }
-        return (int) crc32.getValue();
+        return MessageDigests.toHexString(messageDigest.digest());
```

Comment:
Maybe we should base64 encode these instead to save some bytes?

Reply:
I've changed this to use base64: 17f00d92ee1dc481579188491ff8e53c8c7bd733